### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by jellila** [iquasere/reCOGnizer#1]
Hi :) i get this error and have no idea where it comes from:

`recognizer.py -o /Users/laura/Desktop/try1 -f /Users/laura/Desktop/BIOINFORMATICS/OLD/Newannotations-PDD/508/Complete/QMA0508.fna
Created /Users/laura/Desktop/try1
/Users/laura/miniconda3/envs/iquasere/bin/Databases/COG0001.smp not found! Retrieving from cdd.tar.gz...
/Users/laura/miniconda3/envs/iquasere/bin/Databases/cdd.tar.gz not found! Downloading...
wget ftp://ftp.ncbi.nih.gov/pub/mmdb/cdd/cdd.tar.gz -P /Users/laura/miniconda3/envs/iquasere/bin/Databases
Traceback (most recent call last):
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 477, in <module>
    main()
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 391, in main
    download_resources(args.resources_directory) 
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 123, in download_resources
    run_command('wget ftp://ftp.ncbi.nih.gov/pub/mmdb/cdd/cdd.tar.gz -P {}'.format(database_directory))
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 84, in run_command
    subprocess.run(bashCommand.split(), stdout = stdout)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/subprocess.py", line 423, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'wget': 'wget'
(iquasere) laura@192-168-1-111 ~ % recognizer.py -o /Users/laura/Desktop/try1 -f /Users/laura/Desktop/BIOINFORMATICS/OLD/Newannotations-PDD/508/Complete/QMA0508.fna -rd /Users/laura/Desktop/data 
/Users/laura/Desktop/data/COG0001.smp not found! Retrieving from cdd.tar.gz...
/Users/laura/Desktop/data/cdd.tar.gz not found! Downloading...
wget ftp://ftp.ncbi.nih.gov/pub/mmdb/cdd/cdd.tar.gz -P /Users/laura/Desktop/data
Traceback (most recent call last):
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 477, in <module>
    main()
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 391, in main
    download_resources(args.resources_directory) 
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 123, in download_resources
    run_command('wget ftp://ftp.ncbi.nih.gov/pub/mmdb/cdd/cdd.tar.gz -P {}'.format(database_directory))
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 84, in run_command
    subprocess.run(bashCommand.split(), stdout = stdout)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/subprocess.py", line 423, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/subprocess.py", line 729, in __init__
    restore_signals, start_new_session)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'wget': 'wget'`

Also, could you please explain what kind of databases you need to specify?

Thanks.

Laura

**Comment by iquasere** [comment]
Greetings, @jellila! Looks like your environment cannot find ```wget```. Are you running ```reCOGnizer``` in a Linux machine? If so, please run ```conda install -c anaconda -y wget```.

```reCOGnizer``` requires both CDD and StringDB databases. It automatically downloads them to the folder specified with ```--resources-directory```, and I recommend using this option. In the following runs, keep using this argument with the same folder inputed, so ```reCOGnizer``` won't again download those databases

**Comment by jellila** [comment]
Hi, thanks a lot for your reply! My terminal was finally available again and it seems that with wget installed reCOGniser is working.
Thank you 😊

**Comment by iquasere** [comment]
Glad to know it works now! I thought it would be best to not force install wget by default, as in Linux it should always be available, but I might change that in next versions... Good luck with your work!

**Comment by jellila** [comment]
Sorry, now I get another error. Sorry to disturb you.
`/Users/laura/Desktop/try2/COG0001.smp not found! Retrieving from cdd.tar.gz...
tar -xzf /Users/laura/Desktop/try2/cdd.tar.gz --wildcards "COG*.smp"
tar: Option --wildcards is not supported
Usage:
  List:    tar -tf <archive-filename>
  Extract: tar -xf <archive-filename>
  Create:  tar -cf <archive-filename> [filenames...]
  Help:    tar --help
2020-11-08 23:35:52: Checking if database exists for 6 threads.
Already built COG database for [6] threads.
2020-11-08 23:35:52: Replacing spaces for commas
2020-11-08 23:35:52: Running annotation with RPS-BLAST and COG database as reference.
rpsblast -query /Users/laura/Desktop/BUSCO/508/run_vibrionales_odb10/busco_sequences/fna/prokka/PROKKA_10302020.fna -db /Users/laura/Desktop/try2/COG_6_3 /Users/laura/Desktop/try2/COG_6_2 /Users/laura/Desktop/try2/COG_6_5 /Users/laura/Desktop/try2/COG_6_1 /Users/laura/Desktop/try2/COG_6_0 /Users/laura/Desktop/try2/COG_6_4 -out /Users/laura/Desktop/try/cdd_aligned.blast -outfmt 6 -num_threads 6 -max_target_seqs 1
BLAST engine error: Cannot retrieve path to RPS database
2020-11-08 23:35:52: Converting CDD IDs to respective COG IDs.
Traceback (most recent call last):
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 477, in <module>
    main()
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 415, in main
    cddid = args.resources_directory + '/cddid.tbl')
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 216, in cdd2cog
    blast = pd.read_csv(blast, sep = '\t', header = None)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 688, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 454, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 948, in __init__
    self._make_engine(self.engine)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 1180, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 2010, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 540, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file`

**Comment by iquasere** [comment]
Seems like you are running reCOGnizer on MacOS. If so, this might be fixed with ```brew install gnu-tar```, and rerunning recOGnizer. Any news inform here, and I'll respond in some hours. Going to sleep now ahah

**Comment by jellila** [comment]
It still doesn't work (well, I am fairly clumsy).. yes, I am using a Mac 🙂

**Comment by iquasere** [comment]
Could you tell me if the COG databases are empty? Just check any .pn file, if one is empty they all should be. Also, just to confirm, please check if the cddid.tbl file is empty (this has probably been downloaded correctly, so it shouldn't be empty

**Comment by iquasere** [comment]
Sorry for the mess, but reCOGnizer was never tested on MacOS, and there are these subtles differences between the command line utilities

**Comment by iquasere** [comment]
> It still doesn't work (well, I am fairly clumsy).. yes, I am using a Mac 🙂

Does it give the same error: ```--wildcards is not supported``` ?

**Comment by iquasere** [comment]
@jellila I managed to find a MacOS to test this, and it works with ```gtar```, that you installed with the brew command. I'm going to try launch a new version that is MacOS friendly. For now, you can run
```
cd /Users/laura/miniconda3/envs/iquasere/bin/Databases
gtar /Users/laura/miniconda3/envs/iquasere/bin/Databases/cdd.tar.gz --wildcards "COG*.smp"
``` 
and run the recognizer command as previously. Check if that works ;)

**Comment by iquasere** [comment]
reCOGnizer 1.3.1 has been launched, if you would like to test with it :)

**Comment by jellila** [comment]
I am trying again now. Thank you very much for your help!!! I will let you know how it goes. And also, sorry to give you so much trouble 😔

**Comment by jellila** [comment]
It worked for a while and then stopped :( This is what happened:  

`laura@192-168-1-111 ~ % recognizer.py -o /Users/laura/Desktop/try -rd /Users/laura/Desktop/try2 -f /Users/laura/Desktop/BUSCO/508/run_vibrionales_odb10/busco_sequences/fna/prokka/PROKKA_10302020.faa               
Created /Users/laura/Desktop/try2
/Users/laura/Desktop/try2/COG0001.smp not found! Retrieving from cdd.tar.gz...
/Users/laura/Desktop/try2/cdd.tar.gz not found! Downloading...
wget ftp://ftp.ncbi.nih.gov/pub/mmdb/cdd/cdd.tar.gz -P /Users/laura/Desktop/try2
--2020-11-10 09:24:08--  ftp://ftp.ncbi.nih.gov/pub/mmdb/cdd/cdd.tar.gz
           => ‘/Users/laura/Desktop/try2/cdd.tar.gz’
Resolving ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)... 130.14.250.13
Connecting to ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)|130.14.250.13|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/mmdb/cdd ... done.
==> SIZE cdd.tar.gz ... 4173027251
==> PASV ... done.    ==> RETR cdd.tar.gz ... done.
Length: 4173027251 (3.9G) (unauthoritative)

cdd.tar.gz          100%[===================>]   3.89G   870KB/s    in 53m 23s 

2020-11-10 10:17:35 (1.24 MB/s) - Control connection closed.
Retrying.

--2020-11-10 10:32:36--  ftp://ftp.ncbi.nih.gov/pub/mmdb/cdd/cdd.tar.gz
  (try: 2) => ‘/Users/laura/Desktop/try2/cdd.tar.gz’
Connecting to ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)|130.14.250.13|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/mmdb/cdd ... done.
==> SIZE cdd.tar.gz ... 4173027251
File has already been retrieved.
2020-11-10 10:32:39 (0.00 B/s) - ‘/Users/laura/Desktop/try2/cdd.tar.gz’ saved [4173027251]

brew install gnu-tar
==> Downloading https://homebrew.bintray.com/bottles/gnu-tar-1.32.catalina.bottl
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/158cb67ea9e02435d6710
######################################################################## 100.0%
==> Pouring gnu-tar-1.32.catalina.bottle.1.tar.gz
==> Caveats
GNU "tar" has been installed as "gtar".
If you need to use it as "tar", you can add a "gnubin" directory
to your PATH from your bashrc like:

    PATH="/usr/local/opt/gnu-tar/libexec/gnubin:$PATH"
==> Summary
🍺  /usr/local/Cellar/gnu-tar/1.32: 15 files, 1.7MB
gtar -xzf /Users/laura/Desktop/try2/cdd.tar.gz --wildcards "COG*.smp"
/Users/laura/Desktop/try2/cddid.tbl not found! Downloading...
wget ftp://ftp.ncbi.nlm.nih.gov/pub/mmdb/cdd/cddid.tbl.gz -P /Users/laura/Desktop/try2
--2020-11-10 10:33:44--  ftp://ftp.ncbi.nlm.nih.gov/pub/mmdb/cdd/cddid.tbl.gz
           => ‘/Users/laura/Desktop/try2/cddid.tbl.gz’
Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10
Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/mmdb/cdd ... done.
==> SIZE cddid.tbl.gz ... 6527904
==> PASV ... done.    ==> RETR cddid.tbl.gz ... done.
Length: 6527904 (6.2M) (unauthoritative)

cddid.tbl.gz        100%[===================>]   6.22M   873KB/s    in 8.0s    

2020-11-10 10:33:56 (799 KB/s) - ‘/Users/laura/Desktop/try2/cddid.tbl.gz’ saved [6527904]

gunzip /Users/laura/Desktop/try2/cddid.tbl.gz
/Users/laura/Desktop/try2/fun.txt not found! Downloading...
wget ftp://ftp.ncbi.nlm.nih.gov/pub/COG/COG/fun.txt -P /Users/laura/Desktop/try2
--2020-11-10 10:33:57--  ftp://ftp.ncbi.nlm.nih.gov/pub/COG/COG/fun.txt
           => ‘/Users/laura/Desktop/try2/fun.txt’
Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10
Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/COG/COG ... done.
==> SIZE fun.txt ... 1117
==> PASV ... done.    ==> RETR fun.txt ... done.
Length: 1117 (1.1K) (unauthoritative)

fun.txt             100%[===================>]   1.09K  --.-KB/s    in 0s      

2020-11-10 10:34:01 (178 MB/s) - ‘/Users/laura/Desktop/try2/fun.txt’ saved [1117]

/Users/laura/Desktop/try2/whog not found! Downloading...
wget ftp://ftp.ncbi.nlm.nih.gov/pub/COG/COG/whog -P /Users/laura/Desktop/try2
--2020-11-10 10:34:01--  ftp://ftp.ncbi.nlm.nih.gov/pub/COG/COG/whog
           => ‘/Users/laura/Desktop/try2/whog’
Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.10
Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.10|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/COG/COG ... done.
==> SIZE whog ... 2132879
==> PASV ... done.    ==> RETR whog ... done.
Length: 2132879 (2.0M) (unauthoritative)

whog                100%[===================>]   2.03M   440KB/s    in 4.7s    

2020-11-10 10:34:10 (440 KB/s) - ‘/Users/laura/Desktop/try2/whog’ saved [2132879]

2020-11-10 00:34:10: Checking if database exists for 6 threads.
Generating COG databases for [6] threads.
makeprofiledb -in /Users/laura/Desktop/try2/COG_6_0.pn -title /Users/laura/Desktop/try2/COG_6_0 -out /Users/laura/Desktop/try2/COG_6_0
makeprofiledb -in /Users/laura/Desktop/try2/COG_6_1.pn -title /Users/laura/Desktop/try2/COG_6_1 -out /Users/laura/Desktop/try2/COG_6_1
makeprofiledb -in /Users/laura/Desktop/try2/COG_6_2.pn -title /Users/laura/Desktop/try2/COG_6_2 -out /Users/laura/Desktop/try2/COG_6_2
makeprofiledb -in /Users/laura/Desktop/try2/COG_6_3.pn -title /Users/laura/Desktop/try2/COG_6_3 -out /Users/laura/Desktop/try2/COG_6_3
makeprofiledb -in /Users/laura/Desktop/try2/COG_6_4.pn -title /Users/laura/Desktop/try2/COG_6_4 -out /Users/laura/Desktop/try2/COG_6_4
makeprofiledb -in /Users/laura/Desktop/try2/COG_6_5.pn -title /Users/laura/Desktop/try2/COG_6_5 -out /Users/laura/Desktop/try2/COG_6_5
2020-11-10 00:34:57: Replacing spaces for commas
2020-11-10 00:34:57: Running annotation with RPS-BLAST and COG database as reference.
rpsblast -query /Users/laura/Desktop/BUSCO/508/run_vibrionales_odb10/busco_sequences/fna/prokka/PROKKA_10302020.faa -db /Users/laura/Desktop/try2/COG_6_3 /Users/laura/Desktop/try2/COG_6_2 /Users/laura/Desktop/try2/COG_6_5 /Users/laura/Desktop/try2/COG_6_1 /Users/laura/Desktop/try2/COG_6_0 /Users/laura/Desktop/try2/COG_6_4 -out /Users/laura/Desktop/try/cdd_aligned.blast -outfmt 6 -num_threads 6 -max_target_seqs 1
2020-11-10 00:36:10: Converting CDD IDs to respective COG IDs.
2020-11-10 00:36:11: Converting COG IDs to EC numbers.
/Users/laura/Desktop/try2/cog2ec.py not found! Downloading...
wget -P /Users/laura/Desktop/try2 https://bitbucket.org/scilifelab-lts/lts-workflows-sm-metagenomics/raw/screening_legacy/lts_workflows_sm_metagenomics/source/utils/cog2ec.py
--2020-11-10 10:36:11--  https://bitbucket.org/scilifelab-lts/lts-workflows-sm-metagenomics/raw/screening_legacy/lts_workflows_sm_metagenomics/source/utils/cog2ec.py
Resolving bitbucket.org (bitbucket.org)... 104.192.141.1
Connecting to bitbucket.org (bitbucket.org)|104.192.141.1|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2642 (2.6K) [text/plain]
Saving to: ‘/Users/laura/Desktop/try2/cog2ec.py’

cog2ec.py           100%[===================>]   2.58K  --.-KB/s    in 0s      

2020-11-10 10:36:12 (26.0 MB/s) - ‘/Users/laura/Desktop/try2/cog2ec.py’ saved [2642/2642]

/Users/laura/Desktop/try2/eggnog4.protein_id_conversion.tsv not found! Downloading...
wget -P /Users/laura/Desktop/try2 http://eggnogdb.embl.de/download/eggnog_4.5/eggnog4.protein_id_conversion.tsv.gz
--2020-11-10 10:36:12--  http://eggnogdb.embl.de/download/eggnog_4.5/eggnog4.protein_id_conversion.tsv.gz
Resolving eggnogdb.embl.de (eggnogdb.embl.de)... 194.94.44.170
Connecting to eggnogdb.embl.de (eggnogdb.embl.de)|194.94.44.170|:80... connected.
HTTP request sent, awaiting response... 302 Moved Temporarily
Location: http://eggnog5.embl.de/download/eggnog_4.5/eggnog4.protein_id_conversion.tsv.gz [following]
--2020-11-10 10:36:13--  http://eggnog5.embl.de/download/eggnog_4.5/eggnog4.protein_id_conversion.tsv.gz
Resolving eggnog5.embl.de (eggnog5.embl.de)... 194.94.44.170
Reusing existing connection to eggnogdb.embl.de:80.
HTTP request sent, awaiting response... 200 OK
Length: 427431421 (408M) [application/octet-stream]
Saving to: ‘/Users/laura/Desktop/try2/eggnog4.protein_id_conversion.tsv.gz’

eggnog4.protein_id_ 100%[===================>] 407.63M  1.06MB/s    in 8m 7s   

2020-11-10 10:44:21 (857 KB/s) - ‘/Users/laura/Desktop/try2/eggnog4.protein_id_conversion.tsv.gz’ saved [427431421/427431421]

gunzip /Users/laura/Desktop/try2/eggnog4.protein_id_conversion.tsv.gz
/Users/laura/Desktop/try2/NOG.members.tsv not found! Downloading...
wget -P /Users/laura/Desktop/try2 http://eggnogdb.embl.de/download/eggnog_4.5/data/NOG/NOG.members.tsv.gz
--2020-11-10 10:44:27--  http://eggnogdb.embl.de/download/eggnog_4.5/data/NOG/NOG.members.tsv.gz
Resolving eggnogdb.embl.de (eggnogdb.embl.de)... 194.94.44.170
Connecting to eggnogdb.embl.de (eggnogdb.embl.de)|194.94.44.170|:80... connected.
HTTP request sent, awaiting response... 302 Moved Temporarily
Location: http://eggnog5.embl.de/download/eggnog_4.5/data/NOG/NOG.members.tsv.gz [following]
--2020-11-10 10:44:29--  http://eggnog5.embl.de/download/eggnog_4.5/data/NOG/NOG.members.tsv.gz
Resolving eggnog5.embl.de (eggnog5.embl.de)... 194.94.44.170
Reusing existing connection to eggnogdb.embl.de:80.
HTTP request sent, awaiting response... 200 OK
Length: 58021645 (55M) [application/octet-stream]
Saving to: ‘/Users/laura/Desktop/try2/NOG.members.tsv.gz’

NOG.members.tsv.gz  100%[===================>]  55.33M  1013KB/s    in 84s     

2020-11-10 10:45:53 (675 KB/s) - ‘/Users/laura/Desktop/try2/NOG.members.tsv.gz’ saved [58021645/58021645]

gunzip /Users/laura/Desktop/try2/NOG.members.tsv.gz
python /Users/laura/Desktop/try2/cog2ec.py -c /Users/laura/Desktop/try2/eggnog4.protein_id_conversion.tsv -m /Users/laura/Desktop/try2/NOG.members.tsv
2020-11-10 00:47:28: Converting COG IDs to KEGG Orthologs.
wget -P /Users/laura/Desktop/try2 https://stringdb-static.org/download/COG.mappings.v11.0.txt.gz
--2020-11-10 10:47:28--  https://stringdb-static.org/download/COG.mappings.v11.0.txt.gz
Resolving stringdb-static.org (stringdb-static.org)... 185.111.60.15
Connecting to stringdb-static.org (stringdb-static.org)|185.111.60.15|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 381358937 (364M) [application/x-gzip]
Saving to: ‘/Users/laura/Desktop/try2/COG.mappings.v11.0.txt.gz’

COG.mappings.v11.0. 100%[===================>] 363.69M   917KB/s    in 6m 29s  

2020-11-10 10:53:59 (957 KB/s) - ‘/Users/laura/Desktop/try2/COG.mappings.v11.0.txt.gz’ saved [381358937/381358937]

gunzip /Users/laura/Desktop/try2/COG.mappings.v11.0.txt.gz
wget -P /Users/laura/Desktop/try2 https://stringdb-static.org/download/protein.info.v11.0.txt.gz
--2020-11-10 10:54:03--  https://stringdb-static.org/download/protein.info.v11.0.txt.gz
Resolving stringdb-static.org (stringdb-static.org)... 185.111.60.15
Connecting to stringdb-static.org (stringdb-static.org)|185.111.60.15|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 571627590 (545M) [application/x-gzip]
Saving to: ‘/Users/laura/Desktop/try2/protein.info.v11.0.txt.gz’

protein.info.v11.0. 100%[===================>] 545.15M   988KB/s    in 10m 15s 

2020-11-10 11:04:20 (907 KB/s) - ‘/Users/laura/Desktop/try2/protein.info.v11.0.txt.gz’ saved [571627590/571627590]

gunzip /Users/laura/Desktop/try2/protein.info.v11.0.txt.gz
grep -E 'K[0-9]{5}$' protein.info.v11.0.txt | 
                         awk '{{if (length($NF) == 6) print $1, $NF}}'
grep: protein.info.v11.0.txt: No such file or directory
awk '{if (length($4) == 7) print $1"	"$4}' COG.mappings.v11.0.txt | 
                         sort | join - /Users/laura/Desktop/try2/string2ko.tsv
awk: can't open file COG.mappings.v11.0.txt
 source line number 1
join: /Users/laura/Desktop/try2/string2ko.tsv: No such file or directory
Traceback (most recent call last):
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 485, in <module>
    main()
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 446, in main
    final_result = cog2ko(final_result, cog2ko = args.resources_directory + '/cog2ko.tsv')
  File "/Users/laura/miniconda3/envs/iquasere/bin/recognizer.py", line 347, in cog2ko
    names = ['StringDB','COG','KO'])
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 688, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 454, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 948, in __init__
    self._make_engine(self.engine)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 1180, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/Users/laura/miniconda3/envs/iquasere/lib/python3.6/site-packages/pandas/io/parsers.py", line 2010, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 382, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 674, in pandas._libs.parsers.TextReader._setup_parser_source
FileNotFoundError: [Errno 2] No such file or directory: '/Users/laura/Desktop/try2/cog2ko.ssv'`

**Comment by iquasere** [comment]
These now are errors on my part. The cog2ko module had not yet been properly tested. I made several small fixes, and version 1.3.2 can already been installed, which has that module fully functioning!

**Comment by iquasere** [comment]
reCOGnizer 1.4 has been released, obtaining information from 8 databases. If some problem persists, don't hesitate raising another issue. This particular one seems to be solved
